### PR TITLE
feat(virtual-host): implement domain restriction

### PR DESCRIPTION
**Pull request by phiz71**
If domain restrictions exists on current environment, virtual hosts must respect them and host become mandatory,

Closes gravitee-io/issues#4308

### CODE REVIEW

**Review by aelamrani — CHANGES_REQUESTED** [review]
Did I miss something?

Capture d’écran 2020-10-24 à 11 54 00

**Comment by phiz71** [comment]
> Did I miss something?
> 
> Capture d’écran 2020-10-24 à 11 54 00

No, a problem in the router. I've fixed it.

**Review by ytvnr — COMMENTED** [review]
some remarks and questions

File: src/management/api/proxy/apiProxy.controller.ts [hunk 1]
```diff
@@ -403,6 +407,69 @@ class ApiProxyController {
...
+    this.domainRegexList = this.domainRestrictions.map(value => new RegExp('\\.?' + value, 'i'));
+  }
+
+  focus(inputId: string) {
+    const input: HTMLInputElement = document.querySelector('#' + inputId);
+    if (input.value) {
+      if (_.includes(this.domainRestrictions, input.value) && !input.value.startsWith('.')) {
+        input.value = '.' + input.value;
+      }
+      for (let i = 0; i < this.domainRegexList.length; i++) {
```
[the remark this answers is not in the document]
Comment (phiz71) [reply src/management/api/proxy/apiProxy.controller.ts:446]: I'm ok with this.

File: src/management/api/proxy/apiProxy.controller.ts [hunk 2]
```diff
@@ -403,6 +407,69 @@ class ApiProxyController {
...
+
+  initDomainRestrictions(data: any) {
+    this.domainRestrictions = data.domainRestrictions;
+
+    if (this.domainRestrictions === undefined) {
+      this.domainRestrictions = [];
+    }
+
+    if (this.domainRestrictions.length === 0) {
+      this.hostPattern = '^((?!-)[A-Za-z0-9-]{1,63}(?<!-)\\.)+[A-Za-z]{2,6}$';
```
[the remark this answers is not in the document]
Comment (phiz71) [reply src/management/api/proxy/apiProxy.controller.ts:431]: OK

File: src/management/api/proxy/apiProxy.controller.ts [hunk 3]
```diff
@@ -403,6 +407,69 @@ class ApiProxyController {
       this.formApi.$setDirty();
     }
   }
+
+  getHostOptions(host: string): string[] {
+    this.domainRegexList.forEach(regex => host = host.replace(regex, ''));
```
[the remark this answers is not in the document]
Comment (phiz71) [reply src/management/api/proxy/apiProxy.controller.ts:414]: I can declare a locale variable.

**Pull request merged**